doc/modules/pipeline.rst

```diff
@@ -215,9 +215,13 @@ FeatureUnion: composite feature spaces
 :class:`FeatureUnion` combines several transformer objects into a new
 transformer that combines their output. A :class:`FeatureUnion` takes
 a list of transformer objects. During fitting, each of these
-is fit to the data independently. For transforming data, the
-transformers are applied in parallel, and the sample vectors they output
-are concatenated end-to-end into larger vectors.
+is fit to the data independently. The transformers are applied in parallel,
+and the feature matrices they output are concatenated side-by-side into a
+larger matrix.
+When you want to apply different transformations to each field of the data,
+producing a homogeneous feature matrix from a heterogeneous data source,
```

Comment:
I'm not sure about the claim that the output is homogeneous, nor that this is a helpful way to explain. Too much jargon.